sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/CheckpointStore.cs

```diff
@@ -74,16 +76,37 @@ public abstract class CheckpointStore
         /// <param name="eventHubName">The name of the specific Event Hub the ownership are associated with, relative to the Event Hubs namespace that contains it.</param>
         /// <param name="consumerGroup">The name of the consumer group the checkpoint is associated with.</param>
         /// <param name="partitionId">The identifier of the partition the checkpoint is for.</param>
-        /// <param name="offset">The offset to associate with the checkpoint, indicating that a processor should begin reading form the next event in the stream.</param>
+        /// <param name="offset">The offset to associate with the checkpoint, indicating that a processor should begin reading from the next event in the stream.</param>
         /// <param name="sequenceNumber">An optional sequence number to associate with the checkpoint, intended as informational metadata.  The <paramref name="offset" /> will be used for positioning when events are read.</param>
         /// <param name="cancellationToken">A <see cref="CancellationToken" /> instance to signal a request to cancel the operation.</param>
         ///
-        public abstract Task UpdateCheckpointAsync(string fullyQualifiedNamespace,
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public virtual Task UpdateCheckpointAsync(string fullyQualifiedNamespace,
                                                    string eventHubName,
                                                    string consumerGroup,
                                                    string partitionId,
                                                    long offset,
                                                    long? sequenceNumber,
-                                                   CancellationToken cancellationToken);
+                                                   CancellationToken cancellationToken) => UpdateCheckpointAsync(fullyQualifiedNamespace, eventHubName, consumerGroup, partitionId, null, new CheckpointStartingPosition(sequenceNumber, null, offset), cancellationToken);
```

Comment:
We'll probably need a new internal constructor on `CheckpointPosition` for this path.   This is the only place we would want to allow for a null sequence number.

Reply:
Since there can't be two (long, long?) constructors I added internal setters instead

Reply:
Great idea... that works!